internal/pkg/fieldtags/analyzer.go

```diff
@@ -91,53 +91,13 @@ func (sp *sourcePatterns) isSource(field *ast.Field) bool {
 	}
 
 	tag := field.Tag.Value
-
-	// TODO: consider refactoring this logic into a regex matcher
-	i := 1 // skip initial quote
-	j := 1
-	for j < len(tag) {
-		for j < len(tag) && tag[j] != ':' {
-			j++
-		}
-		key := tag[i:j]
-		if key == "" {
-			return false
-		}
-
-		i = j + 1 // skip colon
-		if i >= len(tag) {
-			return false
-		}
-		if tag[i] == '\\' {
-			i++
-		}
-		i++ // skip quote
-
-		j = i
-		for j < len(tag) && tag[j] != '"' {
-			// skip escape character
-			if tag[j] == '\\' {
-				j++
-			}
-			j++
-		}
-		value := tag[i:j]
-		if value == "" {
-			return false
-		}
-
-		for _, p := range *sp {
-			// value may be a list of comma separated values
-			if p.key == key && strings.Contains(value, p.value) {
-				return true
-			}
-		}
-
-		i = j + 2 // skip closing quote and space
-		j = i
+	if len(tag) <= 2 {
+		return false
 	}
 
-	return false
+	// Tags are a string literal, beginning and ending with '`' or '"'
```

Comment:
I think it would be good to have that comment higher up in the method, since if I understand correctly it explains why we want to return if `len(tag) <= 2`.